src/read/pe/file.rs

```diff
@@ -496,6 +518,76 @@ impl pe::ImageDosHeader {
     }
 }
 
+impl<'data> RichHeaderInfos<'data> {
+    /// Try to detect a rich header in the current PE file, and locate its [`MaskedRichHeaderEntry`s]
+    fn parse<R: ReadRef<'data>>(data: R, nt_header_offset: u64) -> Option<Self> {
+        const RICH_SEQUENCE: &[u8] = &[0x52, 0x69, 0x63, 0x68]; // "Rich"
+        const CLEARTEXT_MARKER: u32 = 0x536e6144; // little-endian "DanS"
+
+        // Locate the rich header, if any
+        // It ends with the ASCII 'Rich' string, before the NT header
+        // It starts at the start marker (a masked ASCII 'DanS' string)
+        let dos_and_rich_header =
+            match data.read_bytes_at_until_sequence(0, RICH_SEQUENCE, nt_header_offset as usize) {
+                Err(()) => return None,
+                Ok(slice) => slice,
+            };
+
+        let xor_key = match data.read_at::<pe::RichCheckSum>(dos_and_rich_header.len() as u64 + 4) {
+            Err(()) => return None,
+            Ok(key) => key,
+        };
+
+        let mut start_marker: Vec<u8> = U32::<LE>::new(LE, CLEARTEXT_MARKER)
+            .as_slice()
+            .iter()
+            .zip(xor_key.checksum.as_slice())
+            .map(|(a, b)| *a ^ *b)
+            .collect();
```

Comment:
Actually, it might make sense to define something like this in src/pe.rs:
```rust
struct RichHeader {
    signature: U32<LE>,
    padding: [U32<LE>; 3],
}
```
and then this can be something like:
```rust
let header = RichHeader {
    signature: U32::new(LE, RICH_HEADER_SIGNATURE ^ mask),
    padding: [U32::new(LE, mask); 3],
};
```

Reply:
I've changed this to follow your first proposal (creating a specific type made me add kind of a useless complexity, questionable doc comments, and converting the `RichHeader` to a slice felt awkward)